### PR TITLE
client/asset/btc: balance should consider untrusted UTXOs

### PR DESCRIPTION
**Pull request by chappjc**
Resolves https://github.com/decred/dcrdex/issues/1546

### CODE REVIEW

**Comment by buck54321** [comment]
Doesn't this mean that I can no longer fund my wallet and begin trading right away?

**Comment by buck54321** [comment]
Oh. This is only SPV. In that case, the wallet already doesn't know about outside outputs until they already have confs = 1.

**Comment by chappjc** [comment]
That's how it has always worked with bitcoind as a backend.

**Comment by chappjc** [comment]
> Oh. This is only SPV. In that case, the wallet already doesn't know about outside outputs until they already have confs = 1.

It knows about a redeem.  That's the main issue.  You complete a swap for BTC, then try to withdraw it before the redeem is confirmed.  The page says you have x btc available, but it's not available to the coin selection code

File: client/asset/btc/spv.go [hunk 1]
```diff
@@ -611,44 +658,20 @@ func (w *spvWallet) listUnspent() ([]*ListUnspentResult, error) {
...
-					if !errors.Is(err, wallet.ErrNotMine) {
-						w.log.Warnf("FetchInputInfo error: %v", err)
-					}
-					safe = false
-					break
-				}
-			}
-		}
+		// by seeing if we control the inputs of its transaction. TODO: Keep a
+		// cache of our redemption outputs and allow those as safe inputs.
```
Comment (chappjc) [inline client/asset/btc/spv.go:662]: I meant to change this comment.  The TODO here is debatable.  The contract output being spent in the redeem is arguably untrusted.  The other party might still have a claim on it via redeem.  It all depends on network conditions and the lock time on the contract.  I think I'd rather play it safe with these.

**Review by JoeGruffins — APPROVED** [review]

**Review by buck54321 — APPROVED** [review]

**Pull request merged**